onnxruntime/python/tools/transformers/optimizer.py

```diff
@@ -99,13 +100,20 @@ def optimize_by_onnxruntime(
             onnx_model_path, sess_options, providers=["CPUExecutionProvider"], **kwargs
         )
     else:
+        gpu_ep = []
+
+        if torch_version.cuda:
+            gpu_ep.append("CUDAExecutionProvider")
+        elif torch_version.hip:
+            gpu_ep.append("ROCMExecutionProvider")
+
         session = onnxruntime.InferenceSession(
-            onnx_model_path, sess_options, providers=["CUDAExecutionProvider"], **kwargs
+            onnx_model_path, sess_options, providers=gpu_ep, **kwargs
         )
-        assert "CUDAExecutionProvider" in session.get_providers()  # Make sure there is GPU
+        assert any(gpu_ep in session.get_providers() for gpu_ep in ["CUDAExecutionProvider","ROCMExecutionProvider"]) # Make sure there is GPU
```

Comment:
Remove this line since I think session creation will fail if no CUDA or Rocm EP, and line 76 has checked.

Reply:
Tested and removed.  I agree.